src/cmds/scripts/pbs_habitat.in

```diff
@@ -649,13 +643,21 @@ chkenv() {
 }
 
 createdir() {
-	if [ ! -d $1 ]; then
-		if ! mkdir $1 ;then
+	if [ -n "$1" -a ! -d "$1" ]; then
+		if ! mkdir -p "$1"; then
 			echo "*** Could not create $1"
 			exit 1
 		fi
 	fi
-	chmod $2 $1
+	if [ -n "$1" -a -n "$2" ]; then
+		chmod "$2" "$1"
+	fi
+}
+
+createpath() {
+	while read mode dir ;do
+		createdir ${PBS_HOME}/${dir} $mode
```

Comment:
Should there be quotes around "${PBS_HOME}/${dir}", like was added to line 100 of pbs_postinstall.in below?